lib/tenstorrent/bh_arc/pvt.c

```diff
@@ -19,594 +18,114 @@
-}
-
-static void WaitSdifReady(uint32_t status_reg_addr)
-{
-	PVT_CNTL_SDIF_STATUS_reg_u sdif_status;
-
-	do {
-		sdif_status.val = ReadReg(status_reg_addr);
-	} while (sdif_status.f.sdif_busy == 1);
-}
-
-static void SdifWrite(uint32_t status_reg_addr, uint32_t wr_data_reg_addr, uint32_t sdif_addr,
-		      uint32_t data)
-{
-	WaitSdifReady(status_reg_addr);
-	PVT_CNTL_SDIF_reg_u sdif;
-
-	sdif.val = PVT_CNTL_SDIF_REG_DEFAULT;
-	sdif.f.sdif_addr = sdif_addr;
-	sdif.f.sdif_wdata = data;
-	sdif.f.sdif_wrn = 1;
-	sdif.f.sdif_prog = 1;
-	WriteReg(wr_data_reg_addr, sdif.val);
-}
-
-static void EnableAgingMeas(void)
-{
-	PDIpCfg0U ip_cfg0;
-
-	ip_cfg0.val = 0;
-	ip_cfg0.f.oscillator_enable = ALL_AGING_OSC;
-	SdifWrite(PVT_CNTL_PD_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_PD_CMN_SDIF_REG_ADDR, IP_CFG0_ADDR,
-		  ip_cfg0.val);
-}
-
-/* setup Interrupt and clk configurations, TS, PD, VM IP configurations. */
-/* Enable contiuous mode for TS and VM. For PD, run once mode should be used. */
-static int PVTInit(void)
-{
-	SetPostCode(POST_CODE_SRC_CMFW, POST_CODE_ARC_INIT_STEP5);
-
-	if (IS_ENABLED(CONFIG_TT_SMC_RECOVERY) || !IS_ENABLED(CONFIG_ARC)) {
-		return 0;
-	}
-
-	/* Enable Process + Voltage + Thermal monitors */
-	PVTInterruptConfig();
-	PVTClkConfig();
-
-	/* Configure TS */
-	SdifWrite(PVT_CNTL_TS_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_TS_CMN_SDIF_REG_ADDR, IP_TMR_ADDR,
-		  0x100); /* 256 cycles for TS */
-
-	/* MODE_RUN_0, 8-bit resolution */
-	TSIpCfg0U ts_ip_cfg0 = {.f.run_mode = 0, .f.resolution = 2};
-
-	SdifWrite(PVT_CNTL_TS_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_TS_CMN_SDIF_REG_ADDR, IP_CFG0_ADDR,
-		  ts_ip_cfg0.val);
-	SdifWrite(PVT_CNTL_TS_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_TS_CMN_SDIF_REG_ADDR, IP_CNTL_ADDR,
-		  0x108); /* ip_run_cont */
-
-	/* Configure PD */
-	SdifWrite(PVT_CNTL_PD_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_PD_CMN_SDIF_REG_ADDR, IP_TMR_ADDR,
-		  0x0); /* 0 cycles for PD */
-	SdifWrite(PVT_CNTL_PD_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_PD_CMN_SDIF_REG_ADDR, IP_CNTL_ADDR,
-		  0x100); /* ip_auto to release reset and pd */
-	EnableAgingMeas();
-
-	/* Configure VM */
-	SdifWrite(PVT_CNTL_VM_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_VM_CMN_SDIF_REG_ADDR, IP_TMR_ADDR,
-		  0x40); /* 64 cycles for VM */
-	SdifWrite(PVT_CNTL_VM_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_VM_CMN_SDIF_REG_ADDR, IP_CFG0_ADDR,
-		  0x1000); /* use 14-bit resolution, MODE_RUN_0, select supply check */
-	SdifWrite(PVT_CNTL_VM_CMN_SDIF_STATUS_REG_ADDR, PVT_CNTL_VM_CMN_SDIF_REG_ADDR, IP_CNTL_ADDR,
-		  0x108); /* ip_auto to release reset and pd */
-
-	/* Wait for all sensors to power up, TS takes 256 ip_clk cycles */
-	Wait(100 * WAIT_1US);
-
-	return 0;
-}
-SYS_INIT_APP(PVTInit);
-
-static uint32_t get_pvt_addr(PvtType type, uint32_t id, uint32_t base_addr)
-{
-	uint32_t offset;
-
-	if (type == VM) {
-		offset = VM_OFFSET;
-	} else {
-		offset = TS_PD_OFFSET;
-	}
-	return id * offset + base_addr;
-}
+#ifdef CONFIG_DT_HAS_TENSTORRENT_BH_PVT_ENABLED
```

Comment:
Same here - I wonder if it's possible to use like `__maybe_unused` or something like that instead of `#ifdef`s.